java/lang/os_command_injection.yml

```diff
@@ -0,0 +1,90 @@
+imports:
+  - java_shared_lang_instance
+patterns:
+  - pattern: |
+      $<RUNTIME_INIT>.exec($<...>$<DYNAMIC_INPUT>$<...>);
+    filters:
+      - variable: RUNTIME_INIT
+        detection: java_lang_os_command_injection_runtime
+      - variable: DYNAMIC_INPUT
+        detection: java_lang_os_command_injection_dynamic_input
+        scope: result
+  - pattern: |
+      new $<PROCESS_BUILDER>($<...>$<DYNAMIC_INPUT>$<...>);
+    filters:
+      - variable: PROCESS_BUILDER
+        regex: \A(java\.lang\.)?ProcessBuilder\z
+      - variable: DYNAMIC_INPUT
+        detection: java_lang_os_command_injection_dynamic_input
+  - pattern: |
+      $<PROCESS_BUILDER_INIT>.command($<...>$<DYNAMIC_INPUT>$<...>);
+    filters:
+      - variable: PROCESS_BUILDER_INIT
+        detection: java_shared_lang_instance
+        filters:
+          - variable: JAVA_SHARED_LANG_INSTANCE_TYPE
+            regex: \A(java\.lang\.)?ProcessBuilder\z
+      - variable: DYNAMIC_INPUT
+        detection: java_lang_os_command_injection_dynamic_input
+auxiliary:
+  - id: java_lang_os_command_injection_runtime
+    patterns:
+      - pattern: $<JAVA_LANG_RUNTIME>.getRuntime();
+        filters:
+          - variable: JAVA_LANG_RUNTIME
+            regex: \A(java\.lang\.)?Runtime\z
+      - pattern: $<JAVA_LANG_RUNTIME>;
+        filters:
+          - variable: JAVA_LANG_RUNTIME
+            detection: java_shared_lang_instance
+            scope: cursor
+            filters:
+              - variable: JAVA_SHARED_LANG_INSTANCE_TYPE
+                regex: \A(java\.lang\.)?Runtime\z
+  - id: java_lang_os_command_injection_process_builder
+    patterns:
+      - pattern: $<JAVA_LANG_PROCESS_BUILDER_INIT>;
+        filters:
+          - variable: JAVA_LANG_PROCESS_BUILDER_INIT
+            detection: java_shared_lang_instance
+            scope: cursor
+            filters:
+              - variable: JAVA_SHARED_LANG_INSTANCE_TYPE
+                regex: \A(java\.lang\.)?ProcessBuilder\z
+  - id: java_lang_os_command_injection_dynamic_input
+    patterns:
+      - pattern: $<USER_INPUT>;
+        filters:
+          - variable: USER_INPUT
+            detection: java_shared_lang_instance
```

Comment:
You probably want `scope: cursor` here. At the moment it's the default `nested` behaviour both here and in the parent patterns, which means it will look at combinations of nodes unnecessarily.